packages/main/src/plugin/filesystem-monitoring.spec.ts

```diff
@@ -0,0 +1,160 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import type { PathLike } from 'node:fs';
+import { existsSync } from 'node:fs';
+
+import { FSWatcher, watch } from 'chokidar';
+import { beforeEach, expect, test, vi } from 'vitest';
+
+import { FileSystemWatcherImpl } from './filesystem-monitoring.js';
+import { Uri } from './types/uri.js';
+
+beforeEach(() => {
+  vi.mock('chokidar');
```

Comment:
maybe we should not mock chokidar to make some real test on it ?

Reply:
For which reason? We generally try to avoid testing dependencies on unit tests

Reply:
To ensure it is working, or have a real regression test

Reply:
I think I'm 50-50

I like to have a unit test with the real implementation, so when we have dependabot, we check if it'll break or not the update of the dependency with the unit tests

But we still to mock the dependency for most of the tests

Reply:
I would be ok to create integration tests (using real chokidar/fs) in addition to the unit tests I have created in this PR, to cover the library/fs related behaviour.

Because `vi.mock('...')` is hoisted to the top of the file, it is not possible to mix tests mocking and not mocking a given library in a same spec file. Is it ok to create some `filesystem-monitoring.integration.spec.ts` in the same directory, or do want to place these tests somewhere else?

Reply:
I am okey having another file for some kind integration, it would improve how we deal with external library if we are able to make this pattern for other package

Reply:
I would add that if we can write a unit test without mock chokidar in a simple way (let's say simpler than with mocking instructions) we should choose that

Reply:
the tests are now using the original chokidar